consul/client.go

```diff
@@ -328,50 +321,34 @@ func (c *Client) localEvent(event serf.UserEvent) {
 
 // RPC is used to forward an RPC call to a consul server, or fail if no servers
 func (c *Client) RPC(method string, args interface{}, reply interface{}) error {
-	// Check the last rpc time
-	var server *serverParts
-	if time.Now().Sub(c.lastRPCTime) < clientRPCCache {
-		server = c.lastServer
-		if server != nil {
-			goto TRY_RPC
-		}
-	}
-
-	// Bail if we can't find any servers
-	c.consulLock.RLock()
-	if len(c.consuls) == 0 {
-		c.consulLock.RUnlock()
+	server := c.serverMgr.FindServer()
+	if server == nil {
+		c.logger.Printf("[ERR] consul: No healthy servers found in the server config")
```

Comment:
This was something I was concerned about as well.  For log entries that are very useful in an incident, but also have a high probability of spamming the logs, the direction I want to go is to introduce a stateful logger and telemetry probe that we can use which will govern the rate of emitted logs so we record the number of times this condition was met and then log only once every 60s.